posts/a-beginners-guide-to-influxdb.md

```diff
@@ -0,0 +1,109 @@
+
+ - Start with importing the required modules in your program
+
+    ``` python
+    from influxdb_client import InfluxDBClient, Point, WritePrecision
+    from influxdb_client.client.write_api import SYNCHRONOUS 
+    ```
+
+ - Then create an InfluxDB client (mention your url, token and org as below) and a writer object
+
+    ```python
+    url = "<my-url>"
+    token = "<my-token>"
+    client = InfluxDBClient(url=url, token=token, org=org)
+    write_api = client.write_api(write_options=SYNCHRONOUS)
+    ```
+
+ - Use the module ‘Point’ that you have imported from influxes_client to convert your data to point, or alternatively create a point yourself
+
+    ```python
+    # Using built-in functionality to create Point
+    p = Point("my_measurement").tag("location", "location1").field("temperature", 25).time(datetime.utcnow(),WritePrecision.NS)
+    
+    # Creating the point manually
+    measurement = "my_measurement"
+    tag_set = "location=location1"
+    field_set = "temperature=25"
+    date_in_nano = str(int(datetime.datetime.utcnow().timestamp() * 1000000000))
+    p = f"{measurement},{tag_set} {field_set} {date_in_nano}" 
+    ```
+ - Finally, write the point to a database. 
+
+    ```python
+    org = "<my-org>"
+    bucket = "<my-bucket>"
+    write_api.write(bucket, org, p) 
+    ```
+
+Here the bucket is a location where the data is stored. It has a retention policy and org attached to it. Retention policy consists of duration, replication factor and shard group duration. The default values for these are infinite duration, one and seven days respectively.
+
+
+# Querying Data from InfluxDB
+Data stored in the InfluxDB can be queried either using a table structure or using a stream. 
+
+ - Again start by importing the required modules and creating an influxes client as well as query object
+
+    ```python
+    from influxdb_client import InfluxDBClient
+
+    url = "<my-url>"
+    token = "<my-token>"
+    client = InfluxDBClient(url=url, token=token, org=org) 
+    query_api = client.query_api() 
+    ```
+ - You can query InfluxDB using a table structure, a sample code below exaplains how to do it.
+
+    ``` python
+    # query using table structure
+    tables = query_api.query('from(bucket:"my-bucket") |> range(start: -10m)')
+    for table in tables: 
+        print(table) 
+        for record in table.records: 
+            print(record.values) 
+    ```
+
+ - You can use below sample code for querying using stream
+
+    ```python
+    # Query using stream
+    records = query_api.query_stream('from(bucket:"my-bucket") |> range(start: -10m)')
+    for record in records: 
+        print(f'Temperature at {record["location"]} - {record["_value"]}')
+    ```
+
+For more information on Influx query language, please refer it's [documentation](https://docs.influxdata.com/influxdb/v1.8/query_language/spec/).
```

Comment:
For more information on Influx query language, please refer to its [documentation](https://docs.influxdata.com/influxdb/v1.8/query_language/spec/).